packages/bson-transpilers/codegeneration/CodeGenerationVisitor.js

```diff
@@ -116,7 +116,12 @@ module.exports = (ANTLRVisitor) => class CodeGenerationVisitor extends ANTLRVisi
         );
       })
       .reduce((obj, c) => {
-        obj[c] = this.Imports[c].template(this.requiredImports[c]);
+        if (c === 'driver') {
+          obj[c] = this.Imports[c].template(mode);
+        } else {
+          obj[c] = this.Imports[c].template(this.requiredImports[c]);
+        }
+
         return obj;
       }, {});
```

Comment:
It might be a good idea to refactor this entire statement to a `for ... of`  loop, just for readability…

Reply:
Replaced with `for .. in`.